### DISCUSSION TITLE
How to make a transaction in next month budget?

### DISCUSSION THREAD

**Discussion by arn4** [firefly-iii/firefly-iii#6547] [Firefly III]
Hello!

I'm trying to figure out how to use budgets effectively. I have a monthly budget for travels and it often happens that  I buy tickets a month or more in advance. I would like that the transaction appears in the month where I spent the money, while it goes in the budget of the month I'm actually travelling. 
I've tried with interest data, but it does not work... How can I achieve this?

**Comment by JC5** [comment]
The date of the transaction is used for the month that it falls in.

**Reply by arn4** [reply]
Ok, but how can the fact that should go in next month budget?
Are there any solution in FireFly?

**Reply by arn4** [reply]
For example, let's say that I have a budget of 100 for travelling each month.
On 20th of October I buy a ticket for the 10th of November costing 20. Meanwhile I increase the Travel Budget of October by 20 and decrease the November one by 20. Is this workaround good?
My expected behaviour is that now my November Travel Budget is -20; on 1st November the cronjob increases by 100 the November Budget, resulting in a budget of 80 (because 20 of them were transferred and spent in October). Is this how it actually works? Do you think there is something better I can do?

**Reply by JC5** [reply]
These kinds of manual updates you have to set yourself. Firefly III won't try to guess that money belongs in the next month or the previous one.

**Reply by arn4** [reply]
> These kinds of manual updates you have to set yourself. Firefly III won't try to guess that money belongs in the next month or the previous one.

Ok I see. I'm not asking for new features. If you know, is the workaround I've describer above a solution? 
More precisely, if I put a negative budget in a future month, when the cronjob acts at the beginning of that month does it add or overwrite the value?

**Reply by JC5** [reply]
I'm not sure. I believe it won't accept a negative and if you set a budget by hand it will not be overwritten. But I can't check for you right I'm afraid, I have no access to my pc 😅

**Reply by arn4** [reply]
I can set the budget for an upcoming month as negative. This is good! Now I've only to wait the 1st of November to see if it also update correctly to (original budget for the month) - (the one I've already substracted)

**Reply by JC5** [reply]
That's not supposed to work haha. I'll check it out

**Reply by arn4** [reply]
Maybe, the best thing is that I don't set the budget automatically and I just update it manually every month as I need. 
How would you suggest to properly handle my situation with FireFly?

**Reply by JC5** [reply]
I would still update the budget automatically each month. That's just easier. But since you spend the money this month, make it part of this month's budget. Don't lift it to the next month.
